### PR TITLE
Detect terminated threads correctly

### PR DESCRIPTION
**Pull request by diversenok**
If a thread exits with STATUS_PENDING, Process Hacker assumes it's still alive, while it is not. Fix this behavior by using the `ThreadIsTerminated` information class.

### CODE REVIEW

File: ProcessHacker/hndlprp.c [hunk 1]
```diff
@@ -1100,7 +1100,7 @@ VOID PhpUpdateHandleGeneral(
 
         if (NT_SUCCESS(status))
         {
-            NTSTATUS exitStatus = STATUS_PENDING;
+            BOOL isTerminated = FALSE;
```
Comment (dmex) [inline ProcessHacker/hndlprp.c]: NT doesn't use the Win32 BOOL type.

File: ProcessHacker/hndlprp.c [hunk 2]
```diff
@@ -1111,10 +1111,34 @@ VOID PhpUpdateHandleGeneral(
                 PhDereferenceObject(name);
             }
 
+            NtQueryInformationThread(dupHandle,
+                ThreadIsTerminated,
+                &isTerminated,
+                sizeof(BOOL),
```
Comment (dmex) [inline ProcessHacker/hndlprp.c]: sizeof(ULONG)

Comment (Biswa96) [reply ProcessHacker/hndlprp.c]: Wouldn't it be easier to use `sizeof isTerminated`? One has to just change the variable declaration.

Comment (diversenok) [reply ProcessHacker/hndlprp.c]: @Biswa96, For some reason, this approach is hardly ever used throughout the codebase. Don't know why, though.

**Comment by diversenok** [comment]
Done.

File: ProcessHacker/hndlprp.c (hunk 2 above)
Comment (dmex) [reply ProcessHacker/hndlprp.c]: @Biswa96 @diversenok It makes it easier for anyone (even without experience) to review the calls 
 and for consistency.

**Pull request merged**